Celeste.Mod.mm/Mod/Helpers/TypeHelper.cs

```diff
@@ -213,37 +208,85 @@ public static class TypeHelper {
             };
         }
 
-        public static void Add_DataNameTypeMap(string name, Type type, bool overwrite = false) {
-            if (overwrite || !entityDataNameToType.ContainsKey(name))
-                entityDataNameToType[name] = type;
+        public readonly static Dictionary<string, Type> FullName_to_Type;
+
+        public readonly static Dictionary<string, Type> EntityDataName_to_Type;
+
+        /// <summary>
+        /// Adds a relationship between an entity Data Name to an object Type. This should *only* be used in situations where you are using a Load method in your CustomEntityAttribute,
+        /// or where you do not use a CustomEntityAttribute at all, but passing the name will always return an object of the same class. 
+        /// </summary>
+        /// <param name="entityDataName">The name of the entity Data, found in your CustomEntityAttribute</param>
+        /// <param name="type">The class that the CustomEntityAttribute would be assigned to.</param>
+        /// <param name="overwrite">Whether or not you want to override previous data added to the Dictionary.</param>
+        public static void Link_DataName_To_Type(string entityDataName, Type type, bool overwrite = false) {
+            if (overwrite || !EntityDataName_to_Type.ContainsKey(entityDataName))
+                EntityDataName_to_Type[entityDataName] = type;
         }
 
-        public static void Add_DataNameTypeMap(Dictionary<string, Type> entityDataNameToType, bool overwrite = false) {
+        /// <summary>
+        /// Appends a Dictionary describing a relationship between an entity Data Name to an object Type. This should *only* be used in situations where you are using a Load method in your CustomEntityAttribute,
+        /// or where you do not use a CustomEntityAttribute at all, but passing the name will always return an object of the same class. 
+        /// </summary>
+        /// <param name="entityDataNameToType">The Dictionary featuring the relationship between names and types.</param>
+        /// <param name="overwrite">Whether or not you want to override previous data added to the Dictionary.</param>
+        public static void Link_DataNames_To_Types(Dictionary<string, Type> entityDataNameToType, bool overwrite = false) {
             foreach(KeyValuePair<string,Type> kvp in entityDataNameToType) {
                 if(overwrite || !entityDataNameToType.ContainsKey(kvp.Key))
                     entityDataNameToType[kvp.Key] = kvp.Value;
             }
```

Comment:
This function seems to do nothing? I think the last line was supposed to write to a different dictionary

Reply:
In practice this code is supposed to do this, so I'll rewrite it as this.
```cs
if(overwrite) entityDataNameToType[kvp.Key] = kvp.Value;
else entityDataNameToType.TryAdd(kvp.Key, kvp.Value);
```